src/plugins/index_pattern_field_editor/public/components/preview/field_preview.tsx

```diff
@@ -36,13 +35,14 @@ export const FieldPreview = () => {
   // To show the preview we at least need a name to be defined, the script or the format
   // and an first response from the _execute API
   const isEmptyPromptVisible =
-    name === null && script === null
+    name === null && script === null && format === null
       ? true
-      : // We have a response from the preview
+      : // If we have some result from the _execute API call don't show the empty prompt
       error !== null || fields.length > 0
       ? false
-      : // We leave it on until we have at least called once the _execute API
-        previewCount === 0;
+      : name === null && format === null
+      ? true
+      : false;
```

Comment:
I wonder if this expression could be made more readable by splitting into several if/else statement with `isEmptyPromptVisible` being `false` by default? I find multiple ternary operators very complex to understand.

Reply:
You right, your suggestion will make it easier to follow. I'll make the change 👍